Modernize `<select>` fallback style

nt1m opened the pull request:
#### 626e6fe843336ebd192a1e5b7f17120ae19118c2
<pre>
Modernize `&lt;select&gt;` fallback style
<a href="https://bugs.webkit.org/show_bug.cgi?id=299251">https://bugs.webkit.org/show_bug.cgi?id=299251</a>
<a href="https://rdar.apple.com/161032297">rdar://161032297</a>

Reviewed by Lily Spiniolas.

Remove outdated aqua background and update the arrow to match the one in RenderThemeCocoa.mm.

* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::MenuListButtonMac::draw):
(WebCore::interpolateGradient): Deleted.
(WebCore::topGradientInterpolate): Deleted.
(WebCore::bottomGradientInterpolate): Deleted.
(WebCore::mainGradientInterpolate): Deleted.
(WebCore::darkTopGradientInterpolate): Deleted.
(WebCore::darkBottomGradientInterpolate): Deleted.
(WebCore::darkMainGradientInterpolate): Deleted.
(WebCore::drawMenuListBackground): Deleted.

Canonical link: <a href="https://commits.webkit.org/300324@main">https://commits.webkit.org/300324@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/97f7b9f3d4ea5cfc03280660b7277008deb442e3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122057 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41759 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128620 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74151 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ad4789af-407c-405f-8b42-40d57f396ef4) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123933 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42474 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50353 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92772 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61654 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/816ddc33-72cf-4e87-8712-ba39637041ff) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125009 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33873 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109302 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73428 "Found 1 new API test failure: TestWTF:WTF_Condition.OneProducerOneConsumerOneSlotTimeout (failure)") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/056b52ec-a328-4cfc-a344-3adabe59bfb6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32886 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27468 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72114 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103380 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27659 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131381 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48996 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37264 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101334 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49370 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105516 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101205 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25673 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46572 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24686 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45719 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48853 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54587 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48323 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51673 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50003 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

lilyspiniolas reviewed (commented):
Since the chevron is the same as the one in RenderThemeCocoa, it would be nice to add a method to the theme for painting the chevron so that it can be reused here rather than duplicating the code.

nt1m committed:
Modernize `<select>` fallback style
https://bugs.webkit.org/show_bug.cgi?id=299251
rdar://161032297

Reviewed by Lily Spiniolas.

Remove outdated aqua background and update the arrow to match the one in RenderThemeCocoa.mm.

* Source/WebCore/SaferCPPExpectations/UnretainedLocalVarsCheckerExpectations:
* Source/WebCore/platform/graphics/mac/controls/MenuListButtonMac.mm:
(WebCore::MenuListButtonMac::draw):
(WebCore::interpolateGradient): Deleted.
(WebCore::topGradientInterpolate): Deleted.
(WebCore::bottomGradientInterpolate): Deleted.
(WebCore::mainGradientInterpolate): Deleted.
(WebCore::darkTopGradientInterpolate): Deleted.
(WebCore::darkBottomGradientInterpolate): Deleted.
(WebCore::darkMainGradientInterpolate): Deleted.
(WebCore::drawMenuListBackground): Deleted.

Canonical link: https://commits.webkit.org/300324@main
